homework/src/index.js

```diff
@@ -30,15 +32,140 @@
+
+    createAndAppend('a', repoTd, { href: data['html_url'], target: '_blank', html: data['name'] });
+
+
+    Object.keys(info).forEach((key) => {
+      const value = info[key];
+      const tr = createAndAppend('tr', infoBody, { id: `${key}Tr`, class: 'rows' });
+      createAndAppend('td', tr, { html: key + ':', class: 'keys' });
+      createAndAppend('td', tr, { html: value, class: 'values' });
+    });
+  }
+
+  //contributors box of the page
+  function createContributorsBox(data, container) {
+
+    const contributorsDiv = createAndAppend('div', container, { id: 'contributorsDiv' });
+
+    const contributorsURL = data['contributors_url'];
+
+    createAndAppend('p', contributorsDiv, { html: 'Contributions' });
+
+
+    fetchJSON(contributorsURL)
+      .catch(err => {
+        createAndAppend('div', contributorsDiv, { html: err.message, class: 'alert-error' });
+      })
+
+      .then(contributors => {
+
+        const contributorsTable = createAndAppend('table', contributorsDiv, { id: 'contributorsTable' });
+
+        const contributorsTbody = createAndAppend('tbody', contributorsTable, { id: 'contributorsTbody' });
+
+
+        contributors.forEach(contributor => {
+
+          const contLink = contributor.html_url;
+          const contImg = contributor.avatar_url;
+          const contName = contributor.login;
+          const contNum = contributor.contributions;
+
+          const contributorsLink = createAndAppend('a', contributorsTbody, { href: contLink, target: '_blank', class: 'contributorsLink' });
+
+          const contributorsTr = createAndAppend('tr', contributorsLink, { id: `${contributor['id']}`, class: 'contributorsTr' });
+
+          const tdImg = createAndAppend('td', contributorsTr, { id: `${contName}Td`, class: 'cont-info' });
+
+          createAndAppend('img', tdImg, { src: contImg, alt: `picture of ${contName}`, class: 'cont-images' });
+
+          createAndAppend('td', contributorsTr, { html: contName, class: 'cont-info cont-name' });
+
+          createAndAppend('td', contributorsTr, { html: contNum, class: 'cont-info cont-num' });
+        });
+      });
+  }
+
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+
+    //blue part of the page
+    const root = document.getElementById('root');
+
+    const headingDiv = createAndAppend('div', root, { id: 'headingDiv' });
+
+    createAndAppend('label', headingDiv, { html: 'HYF Repositories ', class: 'rep-select' });
+
+    const select = createAndAppend('select', headingDiv, { class: 'rep-select', id: 'selectID' });
+
+
+    fetchJSON(url)
+      .catch(err => {
         createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
+      })
+
+      .then(repositories => {
+
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Referring to my previous comment about the wrong placement of `.catch()`, this is was you get in the console when there is the server return an error:

```
Uncaught (in promise) TypeError: Cannot read property 'sort' of undefined
    at fetchJSON.catch.then.repositories (index.js:137)
```